server/jetstream_cluster.go

```diff
@@ -3074,6 +3094,24 @@ func (js *jetStream) processClusterDeleteStream(sa *streamAssignment, isMember,
 		sa.Group.node.Delete()
 	}
 
+	// Cleanup in case account does not exist or node was nil
+	if sacc := s.SystemAccount(); sacc != nil {
```

Comment:
In the happy path this is always handled, this looks a bit strange here.
If the sa.Group.node.Delete() was called from above this is already handled etc.